### PR TITLE
Docstrings for reference tools.meson

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Added Sphinx docstrings to `conan.tools.meson` package.
Docs: https://github.com/conan-io/docs/pull/2533

**Important**: Merge this before `Docs` one.

### CODE REVIEW

File: conan/tools/meson/helpers.py [hunk 1]
```diff
@@ -63,6 +63,12 @@
 
 
 def to_meson_machine(machine_os, machine_arch):
+    """Gets the OS system info as the Meson machine context.
```
Comment (memsharded) [inline conan/tools/meson/helpers.py:66]: These helpers are not public, so make sure they are not documented

File: conan/tools/meson/toolchain.py [hunk 2]
```diff
@@ -272,11 +304,21 @@ def _context(self):
 
     @property
     def content(self):
+        """
+        Gets content of the file to be used by Meson as its context.
```
Comment (memsharded) [inline conan/tools/meson/toolchain.py:308]: Not public interface, the only one atm should be ``generate()``?

**Review by memsharded — APPROVED** [review]

File: conan/tools/meson/helpers.py (hunk 1 above)
Comment (franramirez688) [reply conan/tools/meson/helpers.py:66]: oh, ok. I'll remove them

File: conan/tools/meson/toolchain.py (hunk 2 above)
Comment (franramirez688) [reply conan/tools/meson/toolchain.py:308]: Yeah, I think so. Let's make it private then.

**Review by lasote — APPROVED** [review]

**Pull request merged**